### PR TITLE
Added support for reindexing of dataExpiration

### PR DESCRIPTION
**Pull request by MadhuNEC**
Fixes issue #2160 
Now, we can change dataExpiration value in agent.conf file to get it reflected in database.

### CODE REVIEW

**Comment by MadhuNEC** [comment]
Hi @fgalan ,

If PR seems ok, please merge the PR.
Thanks

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java [hunk 1]
```diff
@@ -123,19 +123,15 @@ public void createCollection(String dbName, String collectionName, long dataExpi
...
-        } // try catch
-        try {
-            if (dataExpiration != 0) {
-                keys = new BasicDBObject().append("_id.origin", 1);
-                options = new IndexOptions().name("cyg_agg_exp").expireAfter(dataExpiration, TimeUnit.SECONDS);
-                db.getCollection(collectionName).createIndex(keys, options);
-            } // if
+            createIndex(db, collectionName, keys, options);
         } catch (Exception e) {
             LOGGER.warn("Error in collection " + collectionName + " creating index ex=" + e.getMessage());
```
Comment (fgalan) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:128]: Should the try/catch be removed (given that createIndex() will deal with the exception internally)?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java [hunk 2]
```diff
@@ -204,20 +200,38 @@ public void createCollection(String dbName, String collectionName, long collecti
                 default:
              }
             options = new IndexOptions().name("cyg_raw_opt");
-            db.getCollection(collectionName).createIndex(keys, options);
+            createIndex(db, collectionName, keys, options);
         } catch (Exception e) {
             LOGGER.warn("Error in collection " + collectionName + " creating index ex=" + e.getMessage());
         } // try catch
```
Comment (fgalan) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:206]: Should the try/catch be removed (given that createIndex() will deal with the exception internally)?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java [hunk 3]
```diff
@@ -204,20 +200,38 @@ public void createCollection(String dbName, String collectionName, long collecti
...
+
+    /**
+     * Create an index for dataExpiration in the given raw collection within the given database.
+     * @param db
+     * @param collectionName
+     * @param keys
+     * @param options
+     * @throws Exception
+     */
+    public void createIndex(MongoDatabase db, String collectionName, BasicDBObject keys,
```
Comment (fgalan) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:222]: Is `throws MongoException` correct?

Given you capture the exception with the try/catch block, I would say that createIndex() is not able to throw any Exception.

File: CHANGES_NEXT_RELEASE [hunk 4]
```diff
@@ -1,9 +1,10 @@
-[cygnus-ngsi] Upgrade Debian version from 11.3 to 11.6 in Dockerfile
-[cygnus-ngsi] Update NameMapping compile log mensage from WARN to ERROR
-[cygnus-common] MongoDB indexes are created depending on DM (#2204)
-[cygnus-common] Upgrade postgresql from 42.4.1 to 42.4.3
-[cygnus-common] OracleSQL backend (#2195)
-[cygnus-common][cygnus-twitter] Upgrade flume-ng-node version from 1.9.0 to 1.11.0 (#2179)
-[cygnus-common][cygnus-twitter][cygnus-ngsi][cygnus-ngsi-ld] Upgrade log4j from v1 (1.2.17) to v2 (2.17.2) series (#1592)
-[cygnus-ngsi] OracleSQL ngsi sink added (for OracleS 11g and 12c) (#2195)
-[cygnus-ngsi] Removes "_" in schema name for DM -schema family (#2201)
+[cygnus-common] Added support for reindexing of dataExpiration (#2160)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]:
```suggestion
[cygnus-common] Added support for reindexing of dataExpiration (#2160, partially)

```

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 1 above)
Comment (MadhuNEC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:128]: Fixed in 1458daf

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 2 above)
Comment (MadhuNEC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:210]: Fixed in 1458daf

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 3 above)
Comment (MadhuNEC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:229]: Fixed in 1458daf

File: CHANGES_NEXT_RELEASE (hunk 4 above)
Comment (MadhuNEC) [reply CHANGES_NEXT_RELEASE]: Fixed in ffafa96

**Comment by MadhuNEC** [comment]
Hi @fgalan ,

Thanks for your support.
I have updated all the 4 suggestions.

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 1 above)
Comment (fgalan) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:128]: It seems the try/catch block is still there... or maybe I'm missing something?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 2 above)
Comment (fgalan) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:210]: It seems the try/catch block is still there... or maybe I'm missing something?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java [hunk 5]
```diff
@@ -204,21 +204,42 @@ public void createCollection(String dbName, String collectionName, long collecti
...
+            createIndex(db, collectionName, keys, options);
         } catch (Exception e) {
             LOGGER.warn("Error in collection " + collectionName + " creating index ex=" + e.getMessage());
         } // try catch
         try {
             if (dataExpiration != 0) {
                 keys = new BasicDBObject().append("recvTime", 1);
                 options = new IndexOptions().name("cyg_raw_exp").expireAfter(dataExpiration, TimeUnit.SECONDS);
-                db.getCollection(collectionName).createIndex(keys, options);
+                createIndex(db, collectionName, keys, options);
```
Comment (fgalan) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:215]: Should the try/catch be removed (given that createIndex() will deal with the exception internally)?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 2 above)
Comment (MadhuNEC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:210]: Ok, will remove try/catch.

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 5 above)
Comment (MadhuNEC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java:215]: Ok, will remove try/catch.

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 1 above)
Comment (MadhuNEC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java]: Fixed in b6045a0

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 2 above)
Comment (MadhuNEC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java]: Fixed in b6045a0

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 5 above)
Comment (MadhuNEC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java]: Fixed in b6045a0

**Review by fgalan — APPROVED** [review]
LGTM

Thanks for the contribution!

**Pull request merged**